central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -620,3 +674,20 @@ func (s *serviceImpl) searchComplianceCheckResults(ctx context.Context, parsedQu
 		ScanResults: storagetov2.ComplianceV2ScanResults(scanResults, scanConfigToIDs),
 	}, nil
 }
+
+func (s *serviceImpl) getLastScanTime(ctx context.Context, clusterID string, profileName string) (*types.Timestamp, error) {
+	// Check the Compliance Scan object to get the scan time.
+	scanQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorProfileName, profileName).
+		AddExactMatches(search.ClusterID, clusterID).
+		ProtoQuery()
+	scans, err := s.scanDS.SearchScans(ctx, scanQuery)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to retrieve scan data for cluster %q and profile %q", clusterID, profileName)
+	}
+	// There should only be a single object for a profile/cluster pair
+	if len(scans) != 1 {
+		return nil, errors.Errorf("Unable to retrieve scan data for cluster %q and profile %q", clusterID, profileName)
```

Comment:
why should there be one scan per cluster per profile? dont we need to sort scans by last executed time

Reply:
The way CO works is that they will only execute a profile once on a given cluster.  So the ComplianceOperatorScan object coming in will always be the last time it was executed.  And for profile 1 on cluster 1 there will only be 1 instance of the ComplianceOperatorScan.  Since we do not store history we always have the latest iteration of that object thus we do not need to sort it.

Reply:
Are we handling the case when CO has not run any scan yet

Reply:
also same as above - Better to return this error as errors.Wrapf(errox.NotFound,"Unable to retrieve scan data for result %q", req.GetId())

Reply:
> Are we handling the case when CO has not run any scan yet

That is handled with the != 1 check below.  That should not happen though.  If we have results we should have a scan. If we don't that is indicative of an issue and as such we want to return an error.  

In light of that, I will not wrap this error in not found because it could be we have another data integrity issue and I don't want to write separate errors for each case.